src/gen/conditions.rs

```diff
@@ -3530,3 +3564,104 @@ fn test_assert_ephemeral_wrong_parent() {
         ErrorCode::AssertEphemeralFailed
     );
 }
+
+#[cfg(test)]
+#[rstest]
+// the default expected errors are post soft-fork, when both new rules are
+// activated
+#[case(ASSERT_HEIGHT_ABSOLUTE, None)]
+#[case(ASSERT_HEIGHT_RELATIVE, Some(ErrorCode::EphemeralRelativeCondition))]
+#[case(ASSERT_SECONDS_ABSOLUTE, None)]
+#[case(ASSERT_SECONDS_RELATIVE, Some(ErrorCode::EphemeralRelativeCondition))]
+#[case(ASSERT_MY_BIRTH_HEIGHT, Some(ErrorCode::EphemeralRelativeCondition))]
+#[case(ASSERT_MY_BIRTH_SECONDS, Some(ErrorCode::EphemeralRelativeCondition))]
+#[case(ASSERT_BEFORE_HEIGHT_ABSOLUTE, None)]
+#[case(
+    ASSERT_BEFORE_HEIGHT_RELATIVE,
+    Some(ErrorCode::EphemeralRelativeCondition)
+)]
+#[case(ASSERT_BEFORE_SECONDS_ABSOLUTE, None)]
+#[case(
+    ASSERT_BEFORE_SECONDS_RELATIVE,
+    Some(ErrorCode::EphemeralRelativeCondition)
+)]
+fn test_relative_condition_on_ephemeral(
+    #[case] condition: ConditionOpcode,
+    #[case] mut expect_error: Option<ErrorCode>,
+    #[values(0, ENABLE_ASSERT_BEFORE)] enable_assert_before: u32,
+    #[values(0, NO_RELATIVE_CONDITIONS_ON_EPHEMERAL)] no_rel_conds_on_ephemeral: u32,
```

Comment:
this is a feature of `rstest` which will test every combination of these parameters